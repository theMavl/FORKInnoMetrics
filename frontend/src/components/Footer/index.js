import React from 'react'
import styles from './style.css'

const contacts = [
  {
    question: 'Product, desktop clients questions',
    users: [
      {
        name: 'Ihar Shulhan',
        email: 'i.shulgan@innopolis.ru',
        telegram: 'ihar_shulhan'
      }
    ]
  },
  {
    question: 'Website questions',
    users: [
      {
        name: 'Daria Larionova',
        email: 'd.larionova@innopolis.ru',
        telegram: 'DaryaLari'
      }
    ]
  }
]

class Footer extends React.Component {
  render() {
    return (
        <footer className={styles.footer}>
          <h4>Contact us</h4>
          <div className={styles.questions}>
            {contacts.map(contact =>  (
                <div className={styles.question} key={contact.question}>
                  <h5>{contact.question}:</h5>
                  {contact.users.map(user => (
                    <div className={styles.user} key={user.email}>
                      <span className={styles.name}>
                        {user.name}
                      </span>
                      <span className={styles.userContact}>
                        <i className='material-icons'>email</i>
                        <a href={`mailto:${user.email}`}>{user.email}</a>
                      </span>
                      <span className={styles.userContact}>
                        <i className='material-icons'>near_me</i>
                        <a href={`https://t.me/${user.telegram}`}>@{user.telegram}</a>
                      </span>
                    </div>
                  ))}

                </div>
            ))}
          </div>
        </footer>

    )
  }
}

export default Footer