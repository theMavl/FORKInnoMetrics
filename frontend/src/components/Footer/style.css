.footer {
    background-color: rgba(var(--footer-color), 1);
    color: rgba(255, 255, 255, 0.8);
    min-height: var(--footer-height);
    padding: calc(var(--page-padding) / 2) var(--page-padding);
}

.footer > h4, .questions h5 {
    margin-top: 0;
}

.questions {
    display: flex;
    /*justify-content: space-between;*/
    flex-wrap: wrap;
}

.question {
    padding-right: var(--cell-size);
}

.name {
    display: block;
    margin: 5px 0;
}

.userContact {
    display: block;
}

.userContact > i {
    font-size: inherit;
    vertical-align: middle;
    padding-right: 5px;
}

.userContact > a {
    color: inherit;
}