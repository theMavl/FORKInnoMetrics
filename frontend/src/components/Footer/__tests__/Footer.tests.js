import React from 'react'
import TestRenderer from 'react-test-renderer'
import Footer from '../index'

describe('Test Footer component', () => {

  it('Test render Footer component', () => {

    const instance = TestRenderer.create(
      <Footer/>
    )
    expect(instance.toJSON()).toMatchSnapshot()
  })
})