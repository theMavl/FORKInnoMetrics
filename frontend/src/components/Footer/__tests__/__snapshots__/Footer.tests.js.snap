// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer component Test render Footer component 1`] = `
<footer>
  <h4>
    Contact us
  </h4>
  <div>
    <div>
      <h5>
        Product, desktop clients questions
        :
      </h5>
      <div>
        <span>
          Ihar Shulhan
        </span>
        <span>
          <i
            className="material-icons"
          >
            email
          </i>
          <a
            href="mailto:i.shulgan@innopolis.ru"
          >
            i.shulgan@innopolis.ru
          </a>
        </span>
        <span>
          <i
            className="material-icons"
          >
            near_me
          </i>
          <a
            href="https://t.me/ihar_shulhan"
          >
            @
            ihar_shulhan
          </a>
        </span>
      </div>
    </div>
    <div>
      <h5>
        Website questions
        :
      </h5>
      <div>
        <span>
          Daria Larionova
        </span>
        <span>
          <i
            className="material-icons"
          >
            email
          </i>
          <a
            href="mailto:d.larionova@innopolis.ru"
          >
            d.larionova@innopolis.ru
          </a>
        </span>
        <span>
          <i
            className="material-icons"
          >
            near_me
          </i>
          <a
            href="https://t.me/DaryaLari"
          >
            @
            DaryaLari
          </a>
        </span>
      </div>
    </div>
  </div>
</footer>
`;
